.build/docker/Dockerfile

```diff
@@ -0,0 +1,136 @@
+	rm -rf /var/lib/apt/lists/*; \
+	dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')"; \
+	wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch"; \
+	wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc"; \
+	export GNUPGHOME="$(mktemp -d)"; \
+	gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \
+	gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \
+	gpgconf --kill all; \
+	rm -rf "$GNUPGHOME" /usr/local/bin/gosu.asc; \
+	apt-mark auto '.*' > /dev/null; \
+	[ -z "$savedAptMark" ] || apt-mark manual $savedAptMark > /dev/null; \
+	apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \
+	chmod +x /usr/local/bin/gosu; \
+	gosu --version; \
+	gosu nobody true
+COPY $CASSANDRA_TARBALL cassandra-bin.tgz
+ENV CASSANDRA_HOME /opt/cassandra
+ENV CASSANDRA_CONF /etc/cassandra
+ENV PATH $CASSANDRA_HOME/bin:$PATH
+
+RUN set -eux; \
+	savedAptMark="$(apt-mark showmanual)"; \
+	apt-get update; \
+	apt-get install -y --no-install-recommends ca-certificates dirmngr gnupg wget; \
+	rm -rf /var/lib/apt/lists/*; \
+	\
+	apt-mark auto '.*' > /dev/null; \
+	[ -z "$savedAptMark" ] || apt-mark manual $savedAptMark > /dev/null; \
+	apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \
+	\
+	mkdir -p "$CASSANDRA_HOME"; \
+	tar --extract --file cassandra-bin.tgz --directory "$CASSANDRA_HOME" --strip-components 1; \
+	rm cassandra-bin.tgz*; \
+	\
+	[ ! -e "$CASSANDRA_CONF" ]; \
+	mv "$CASSANDRA_HOME/conf" "$CASSANDRA_CONF"; \
+	ln -sT "$CASSANDRA_CONF" "$CASSANDRA_HOME/conf"; \
+	\
+	dpkgArch="$(dpkg --print-architecture)"; \
+	case "$dpkgArch" in \
+		ppc64el) \
+# https://issues.apache.org/jira/browse/CASSANDRA-13345
+# "The stack size specified is too small, Specify at least 328k"
+			grep -- '^-Xss256k$' "$CASSANDRA_CONF/jvm-server.options"; \
+			sed -ri 's/^-Xss256k$/-Xss512k/' "$CASSANDRA_CONF/jvm-server.options"; \
+			grep -- '^-Xss512k$' "$CASSANDRA_CONF/jvm-server.options"; \
+			;; \
+	esac; \
+	\
+	mkdir -p "$CASSANDRA_CONF" /var/lib/cassandra /var/log/cassandra; \
+	chown -R cassandra:cassandra "$CASSANDRA_CONF" /var/lib/cassandra /var/log/cassandra; \
+	chmod 777 "$CASSANDRA_CONF" /var/lib/cassandra /var/log/cassandra; \
+	chmod -R a+rwX "$CASSANDRA_CONF"; \
+	ln -sT /var/lib/cassandra "$CASSANDRA_HOME/data"; \
+	ln -sT /var/log/cassandra "$CASSANDRA_HOME/logs"; \
+	\
+# smoke test
+	cassandra -v
+
+VOLUME /var/lib/cassandra
+
+COPY docker-entrypoint.sh /usr/local/bin/
+ENTRYPOINT ["docker-entrypoint.sh"]
+
+# 7000: intra-node communication
+# 7001: TLS intra-node communication
+# 7199: JMX
+# 9042: CQL
+# 9160: thrift service
+EXPOSE 7000 7001 7199 9042 9160
+
+# Container is healthy when JMX port is up
+HEALTHCHECK CMD /usr/bin/nc localhost -z 7199 || exit 1
```

Comment:
This might be replaced by nodetool status returning 0 exit code. The fact that this port is up does not necessarily mean that nodetool status will be successful as port is open but the node is still booting anyway under the hood.